cli/ballerina-cli/src/test/java/io/ballerina/cli/cmd/BuildCommandTest.java

```diff
@@ -888,14 +895,228 @@ public void testBuildBalProjectWithCustomMavenRepo() throws IOException {
+        System.setProperty("user.dir", projectPath.toString());
+        BuildCommand buildCommand = new BuildCommand(projectPath, printStream, printStream, false);
+
+        new CommandLine(buildCommand).parseArgs("--sticky");
+        buildCommand.execute();
+        String buildLog = readOutput(true);
+        Assert.assertEquals(buildLog.replaceAll("\r", ""), getOutput("build-new-proj-with-dep.txt"));
+
+        // Dependencies.toml should have the latest versions of the dependencies
+        Path actualDependenciesToml = projectPath.resolve("Dependencies.toml");
+        Path expectedDependenciesToml = testResources.resolve("dep-dist-version-projects").resolve("expected-dep-tomls")
+                .resolve("new-pkg.toml");
+        Assert.assertTrue(actualDependenciesToml.toFile().exists());
+        assertTomlFilesEquals(actualDependenciesToml, expectedDependenciesToml);
+
+        // Delete dependencies.toml from the project
+        Files.delete(actualDependenciesToml);
+        deleteDirectory(projectPath.resolve("target"));
+    }
+
+    @Test(description = "Build a project already built with an older distribution without sticky flag",
+            groups = {"proj-with-deps-update-policy"})
+    public void testBuildProjectPrecompiledWithOlderDistWithoutStickyFlag() throws IOException {
+        Path projectPath = testResources.resolve("dep-dist-version-projects").resolve("preCompiledPackage");
+        replaceDependenciesTomlContent(projectPath, "**INSERT_DISTRIBUTION_VERSION_HERE**", "2201.5.0");
+        System.setProperty("user.dir", projectPath.toString());
+        BuildCommand buildCommand = new BuildCommand(projectPath, printStream, printStream, false);
+
+        new CommandLine(buildCommand).parseArgs();
+        buildCommand.execute();
+        String buildLog = readOutput(true);
+        Assert.assertEquals(
+                buildLog.replaceAll("\r", ""),
+                getOutput("build-old-dist-precomp-proj-with-dep.txt")
+                        .replaceAll("INSERT_NEW_DIST_VERSION_HERE", getNewVersionForOldDistWarning())
+                        .replaceAll("INSERT_OLD_DIST_VERSION_HERE", getOldVersionForOldDistWarning("2201.5.0")));
+
+        // Dependencies.toml should be updated to the latest minor versions.
+        // depA:1.0.0 -> depA:1.1.0
+        // depB:1.0.0 -> depB:1.1.0
+        Path actualDependenciesToml = projectPath.resolve("Dependencies.toml");
+        Path expectedDependenciesToml = testResources.resolve("dep-dist-version-projects").resolve("expected-dep-tomls")
+                .resolve("old-dist-precomp-pkg-with-no-sticky.toml");
+        Assert.assertTrue(actualDependenciesToml.toFile().exists());
+        assertTomlFilesEquals(actualDependenciesToml, expectedDependenciesToml);
+
+        // Revert the distribution version and dependency versions to the placeholder values
+        replaceDependenciesTomlContent(projectPath, RepoUtils.getBallerinaShortVersion(),
```

Comment:
We can improve this to copy the project files to a temporary location and replacing them. This will avoid accidental committing of modified files.

Reply:
Then why do we need to revert the values?

Reply:
That is because we are using the same project for all the tests that are in the group `proj-with-deps-update-policy`. So we are reverting the changes to the test-package copied to the temporary directory after each test belonging to that group. This makes it ready for the next test.

Reply:
Let's improve this to use separate projects for each test to avoid failures in the cleanup. Please create an issue to track it.